sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb

```diff
@@ -0,0 +1,618 @@
+    "```\n",
+    "\n",
+    "We added to following callbacks:\n",
+    "```python\n",
+    "hvd.callbacks.BroadcastGlobalVariablesCallback(0)\n",
+    "hvd.callbacks.MetricAverageCallback()\n",
+    "```\n",
+    "\n",
+    "Configured the optimizer:\n",
+    "```python\n",
+    "opt = Adam(lr=learning_rate * size, decay=weight_decay)\n",
+    "opt = hvd.DistributedOptimizer(opt)\n",
+    "```\n",
+    "and choose to save checkpoints and send TensorBoard logs only from the ```python hvd.rank() == 0``` instance\n",
+    "\n",
+    "To start an distributed training job with Horovod, you'll need to configure the job distribution:\n",
+    "```python\n",
+    "distributions = {'mpi': {\n",
+    "                    'enabled': True,\n",
+    "                    'processes_per_host': # Number of Horovod processes per host\n",
+    "                        }\n",
+    "                }\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "distributions = {'mpi': {\n",
+    "                    'enabled': True,\n",
+    "                    'processes_per_host': 4\n",
+    "                        }\n",
+    "                }\n",
+    "hyperparameters = {'epochs': 60, 'batch-size' : 256}\n",
+    "\n",
+    "source_dir = os.path.join(os.getcwd(), 'source_dir')\n",
+    "estimator_dist = TensorFlow(base_job_name='dist-cifar10-tf',\n",
+    "                       entry_point='cifar10_keras_main.py',\n",
+    "                       source_dir=source_dir,\n",
+    "                       role=role,\n",
+    "                       framework_version='1.12.0',\n",
+    "                       py_version='py3',\n",
+    "                       hyperparameters=hyperparameters,\n",
+    "                       train_instance_count=2, train_instance_type='ml.p3.8xlarge',\n",
+    "                       tags = [{'Key' : 'Project', 'Value' : 'cifar10'},{'Key' : 'TensorBoard', 'Value' : 'dist'}],\n",
+    "                       metric_definitions=keras_metric_definition,\n",
+    "                       distributions=distributions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll configure ```wait=False``` if you want to see the output logs change to ```wait=True```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "remote_inputs = {'train' : inputs+'/train', 'validation' : inputs+'/validation', 'eval' : inputs+'/eval'}\n",
+    "estimator_dist.fit(remote_inputs, wait=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Local TensorBoard command\n",
+    "Using TenosrBoard we can compare between the different jobs we ran, the following command will print the tensorboard command.\n",
```

Comment:
"Using TensorBoard we can compare the jobs we ran. The following command prints the TensorBoard..."